firmware/Cargo.toml

```diff
@@ -12,7 +12,7 @@ panic-rtt-target = { version = "0.1.0", features = ["cortex-m"] }
 stm32ral = { version = "0.4.1", features = ["stm32f7x3", "rt"] }
 hs-probe-bsp = { path = "../hs-probe-bsp" }
 usb-device = { version = "0.2.5", features = ["control-buffer-256"] }
-usbd-serial = { path = "../usbd-serial", features = ["high-speed"] }
+usbd-serial = { version = "0.1.1", features = ["high-speed"] }
```

Comment:
This will break as soon as usbd-serial gets updated to 0.1.2 or so.

Reply:
I don't think this will be a problem. As a test, I switched to a local fork of usbd-serial and set the version back to 1.1.0 and it still worked.

### hs-probe-firmware
```diff
diff --git a/Cargo.toml b/Cargo.toml
index c6d1434..dbb6333 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -9 +9 @@ usb-device = { git = "https://github.com/Disasm/usb-device", rev = "5d40bdce478e
-usbd-serial = { git = "https://github.com/Disasm/usbd-serial", rev = "827116e5fab66a51ab7d68774419b0f187b24b90" }
+usbd-serial = { path = "../usbd-serial" }
diff --git a/firmware/Cargo.toml b/firmware/Cargo.toml
index 8f7453d..c56891c 100644
--- a/firmware/Cargo.toml
+++ b/firmware/Cargo.toml
@@ -15 +15 @@ usb-device = { version = "0.2.5", features = ["control-buffer-256"] }
-usbd-serial = { version = "0.1.1", features = ["high-speed"] }
+usbd-serial = { version = "0.1.0", features = ["high-speed"] }
```

### usbd-serial
```diff
diff --git a/Cargo.toml b/Cargo.toml
index 1e311d8..645e9f8 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -4 +4 @@ description = "USB CDC-ACM serial port class for use with usb-device."
-version = "0.1.1"
+version = "0.1.0"
```

Reply:
But if you use git-dependency patch with 0.1.0 it will not work, because upstream usbd-serial is 0.1.1.

Reply:
This still works with https://github.com/GrantM11235/usbd-serial/commit/c2ed4f03aab1df65b2fef63147e11b3463f8963f
```diff
diff --git a/Cargo.toml b/Cargo.toml
index c6d1434..08e4637 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -9 +9 @@ usb-device = { git = "https://github.com/Disasm/usb-device", rev = "5d40bdce478e
-usbd-serial = { git = "https://github.com/Disasm/usbd-serial", rev = "827116e5fab66a51ab7d68774419b0f187b24b90" }
+usbd-serial = { git = "https://github.com/GrantM11235/usbd-serial", rev = "c2ed4f03aab1df65b2fef63147e11b3463f8963f" }
diff --git a/firmware/Cargo.toml b/firmware/Cargo.toml
index 8fd07bf..817652f 100644
--- a/firmware/Cargo.toml
+++ b/firmware/Cargo.toml
@@ -15 +15 @@ usb-device = { version = "0.2.5", features = ["control-buffer-256"] }
-usbd-serial = { version = "0.1.1", features = ["high-speed"] }
+usbd-serial = { version = "0.1.0", features = ["high-speed"] }
```

Reply:
Hmm, this indeed works, thank you for testing!